### PR TITLE
Discard reporter output for tests that succeed.

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: test/test/CompilerTest.scala [hunk 1]
```diff
@@ -234,7 +236,22 @@ abstract class CompilerTest {
...
     val processor = if (allArgs.exists(_.startsWith("#"))) Bench else Main
-    val reporter = processor.process(allArgs)
+    val storeReporter = new Reporter {
+      private val consoleReporter = new ConsoleReporter()
+      private var reporter = new StoreReporter(consoleReporter)
+      def doReport(d: Diagnostic)(implicit ctx: Context): Unit = {
+        if (reporter == null) {
+          consoleReporter.doReport(d)
+        } else {
+          reporter.doReport(d)
```
Comment (DarkDimius) [inline test/test/CompilerTest.scala]: why not wait until test finished and then `if (reporter.hasErrors) reporter.flush`?

Comment (nicolasstucki) [inline test/test/CompilerTest.scala]: Because I do not have access to the context here.

File: test/test/CompilerTest.scala [hunk 2]
```diff
@@ -234,6 +236,21 @@ abstract class CompilerTest {
...
     val allArgs = args ++ defaultOptions
     val processor = if (allArgs.exists(_.startsWith("#"))) Bench else Main
+    val storeReporter = new Reporter {
+      private val consoleReporter = new ConsoleReporter()
+      private var innerStoreReporter = new StoreReporter(consoleReporter)
+      def doReport(d: Diagnostic)(implicit ctx: Context): Unit = {
+        if (innerStoreReporter == null) {
+          consoleReporter.report(d)
+        } else {
+          innerStoreReporter.report(d)
```
Comment (nicolasstucki) [inline test/test/CompilerTest.scala]: Since last update I changed `.doReport(d)` with `.report(d)`.

**Comment by nicolasstucki** [comment]
For the `validate-junit` test, the output reduced from 2,5 MB to 338 KB.

**Comment by odersky** [comment]
LGTM

**Comment by nicolasstucki** [comment]
Just changed the commit message

**Pull request merged**